src/createStore.js

```diff
@@ -210,3 +192,42 @@ export default function createStore(reducer, initialState, enhancer) {
     replaceReducer
   }
 }
+
+/**
+ *
+ * TODO: Docs.
+ *
+ * @param {Function} reducer @see createStore
+ *
+ * @param {any} [initialState] @see createStore
+ *
+ * @param {Function} enhancer The store enhancer. You may optionally specify it
+ * to enhance the store with third-party capabilities such as middleware,
+ * time travel, persistence, etc. The only store enhancer that ships with Redux
+ * is `applyMiddleware()`.
+ *
+ * @returns {Store} An initialized Redux store that lets you read the state,
+ * dispatch actions and subscribe to changes.
+ */
+export default function initializeStore(reducer, initialState, enhancer) {
+  if (typeof initialState === 'function' && typeof enhancer === 'undefined') {
+    enhancer = initialState
+    initialState = undefined
+  }
+
+  if (typeof enhancer !== 'undefined') {
+    if (typeof enhancer !== 'function') {
+      throw new Error('Expected the enhancer to be a function.')
+    }
+  }
+
+  var finalCreateStore = enhancer ? enhancer(createStore) : createStore
```

Comment:
Enhancer can potentially dispatch an action which will be called before `ActionTypes.INIT` therefore this does not ensure that first action has all the middlewares applied.